### DISCUSSION TITLE
Using PyInstaller Splash Displays Imports

### DISCUSSION THREAD

**Discussion by grinchify** [pyinstaller/pyinstaller#7850] [Help] [answered]
![module_print](https://github.com/pyinstaller/pyinstaller/assets/83351519/fc6c1708-605b-4d87-bcf5-56099ae42175)
How can I prevent such statements (import statements that are used in my code by the code itself or another module) from being shown on the splash page? I am not looking for it to be disabled as I do update it within my code, but just looking for it to not display the imports

Using the following expression to compile

python -m PyInstaller --clean --noconfirm grinch_bot.spec

**Comment by rokm** [comment]
In onefile build, splash displays the files it extracts. In the current implementation, there is no way to disable that.

That said, PyInstaller by default should not not collect .pyc files from `__pycache__` - unless you (or a misbehaving hook) call `collect_all` or `collect_data_files` with `include_py_files=True`...

**Reply by asmaier** [reply]
I think this should be documented also at https://pyinstaller.org/en/stable/spec-files.html#PyInstaller.building.splash.Splash.__init__ under `text_default`.  I set `text_default='Loading....'` and was confused, why it didn't show up.

**Comment by grinchify** [comment]
Why isnt there any way to disable that? How can I go about editing the code itself to disable that.

**Reply by rokm** [reply, marked as answer]
> Why isnt there any way to disable that?

Because up until now, nobody took an issue with it.

>  How can I go about editing the code itself to disable that.

Disable this line:
https://github.com/pyinstaller/pyinstaller/blob/a732e7b98b42e5f773420e2b087f2f24531de7b8/bootloader/src/pyi_launch.c#L332

and recompile the bootloader.

**Reply by grinchify** [reply]
Issue still persists, does not seem like it worked.

**Reply by rokm** [reply]
Have you recompiled the bootloader, and then rebuilt your application so that the modifed bootloader is used? (Unless you installed PyInstaller from source in editable mode, you will likely need to reinstall it after compiling the bootloader).

**Reply by grinchify** [reply]
How do I check / know if the modified bootloader is used?

I did the following steps

git clone https://github.com/pyinstaller/pyinstaller
( I renamed the pyinstaller file)
cd pyinstaller_raw_code, then cd bootloader/src/pyi_launch.c
Commented out pyi_splash_update_prg(splash_status, ptoc);
python ./waf distclean all
pip install .

**Reply by grinchify** [reply]
Looks like the distclean all raises some errors and outputs a config file. Looking into it, give me a few minutes

**Reply by grinchify** [reply]
Ok, Thanks a lot. Its working fine. My bootloader compilation had issues ( I had to install visual studio to fix it ). On top of that had to replace the PyInstallers bootloader run exes with the one in my pyinstaller_raw_code one (the one where I compiled the bootloader from)
